src/db.rs

```diff
@@ -119,12 +123,22 @@ impl DbInner {
         // Since the key remains unchanged during the point query, we only need to compute
         // the hash value once and pass it to the filter to avoid unnecessary hash computation
         let key_hash = filter::filter_hash(key);
+        let key_bytes = Bytes::copy_from_slice(key);
```

Comment:
We had that discussion about whether the range to `scan` should be by reference (`scan(&..)` vs `scan(..)`). Since we decided to pass by value, we need `DbIterator` to own the range value (through `SstIterator`) so that it doesn't get dropped when `scan` returns. We might be able to do the same `AsRef` trick that is used for `SsTableHandle`. I will give it a shot. If there is a more straightforward approach, let me know. I'm still pretty new to rust.

Reply:
@criccomini I think a better approach might be to pull the range/key out of `SstIterator`. I have a draft here which gives the idea: https://github.com/hachikuji/slatedb/blob/range-scans-externalize-sst-iter-range/src/sst_iter.rs#L289. If this looks reasonable, we might want to consider moving it to a follow-up since this patch is already quite large.

Reply:
Ah, nice. Yea, this looks good. I agree that this should be implemented as a follow-on. As you said, the current PR is pretty lengthy.